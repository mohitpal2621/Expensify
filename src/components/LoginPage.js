import React from "react";
import { connect } from "react-redux";
import { startLogin } from "../actions/auth";

const LoginPage = (props) => (
    <div>
        <button onClick={props.startLogin}>LOGIN</button>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);