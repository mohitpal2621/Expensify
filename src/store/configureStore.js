import { combineReducers, configureStore } from "@reduxjs/toolkit";
import expenseReducer from '../reducers/expenseReducer';
import filtersReducer from '../reducers/filterReducer';
import authReducer from "../reducers/authReducer";

const rootreducer = combineReducers({
    expenses: expenseReducer,
    filters: filtersReducer,
    auth: authReducer
});

export default () => {
    const store = configureStore({
        reducer: rootreducer
    });

    return store;
};
