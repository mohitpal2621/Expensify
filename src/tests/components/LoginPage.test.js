import React from "react"; 
import {render, fireEvent} from "@testing-library/react";
import "@testing-library/jest-dom";
import LoginPage from "../../components/LoginPage";
import { Provider } from "react-redux";
import configureStore from "../../store/configureStore";

const store = configureStore();

jest.mock('firebase/auth', () => ({
    GoogleAuthProvider: jest.fn(() => ({
      setCustomParameters: jest.fn(),
    })),
    getAuth: jest.fn(() => ({})),
}));

test("Should render the login page correctly", () => {
    const {container} = render(<Provider store={store}><LoginPage/></Provider>)
    expect(container).toMatchSnapshot();
});

// test("Should call startLogin on button click", () => {
//   const startlogin = jest.fn();
//   const {getByText} = render(<Provider store={store}><LoginPage startLogin={startlogin}/></Provider>)
//   const btn = fireEvent.click(getByText("LOGIN"));
//   expect(startlogin).toHaveBeenCalled();
// })