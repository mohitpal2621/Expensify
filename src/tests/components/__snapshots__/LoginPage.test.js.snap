// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the login page correctly 1`] = `
<div>
  <div>
    <button>
      LOGIN
    </button>
  </div>
</div>
`;
